fud/fud/stages/xilinx/execution.py

```diff
@@ -75,21 +60,56 @@ def run(xclbin: SourceType.Path) -> SourceType.String:
                         f"user_pre_sim_script={new_dir.name}/pre_sim.tcl\n"
                     )
                     xrt_ini_config.append(
-                        f"user_post_sim_script={new_dir.name}/pre_sim.tcl\n"
+                        f"user_post_sim_script={new_dir.name}/post_sim.tcl\n"
                     )
                 f.writelines(xrt_ini_config)
 
             # Extra Tcl scripts to produce a VCD waveform dump.
             if waveform:
                 with open("pre_sim.tcl", "w") as f:
-                    f.writelines([
-                        "open_vcd\n",
-                        "log_vcd *\n",
-                    ])
+                    f.writelines(
+                        [
+                            "open_vcd\n",
+                            "log_vcd *\n",
+                        ]
+                    )
                 with open("post_sim.tcl", "w") as f:
-                    f.writelines([
-                        "close_vcd\n",
-                    ])
+                    f.writelines(
+                        [
+                            "close_vcd\n",
+                        ]
+                    )
+
+            os.environ["XRT_INI_PATH"] = f"{new_dir.name}/xrt.ini"
```

Comment:
Hella weird that putting the line here was the key to making PYNQ pay attention to the setting, as mentioned in https://github.com/cucapra/calyx/issues/1127#issuecomment-1200384844! It's hard to think of why this would make the difference, but on the other hand, this organization is pretty clear and straightforward (more so than the old version) anyway, so let's roll with it and not look too closely.